### PR TITLE
feat(core-gateway): restore access-log fan-out to lightbridge-authz-usage

### PR DESCRIPTION
**Pull request by stephane-segning**
## 1. Summary

This PR changes:

- `charts/core-gateway/templates/otel.yaml`: restores a dedicated `-usage` `OpenTelemetryCollector`, removed in `948814de` ("remove usage-collector + otel-operator (external)"), that receives Envoy's access-log OTLP stream and fans it out to **both** Grafana Alloy (unchanged, → Loki, ADR-0005 per-user observability) **and** a new `otlphttp/lightbridge_usage` exporter hitting `lightbridge-authz-usage`'s `/v1/otel/logs` ingest.
- `charts/core-gateway/templates/envoy-proxy.yaml`: points the `EnvoyProxy` `telemetry.accessLog` OTLP sink at the new collector instead of straight at Alloy.
- `charts/core-gateway/templates/certificate-usage-ca.yaml` (new): a same-namespace `cert-manager` `Certificate`, off the same `self-signed-ca` `ClusterIssuer` that signs `lightbridge-authz-usage`'s own TLS cert, so the collector can verify that service's HTTPS listener with a real CA — no `insecure: true`/skip-verify.
- `charts/core-gateway/values.yaml`: adds the `otelCollector.usageCa.{issuer,issuerKind}` knob backing the new Certificate.

It solves:

- `lightbridge-authz-usage` (crates/lightbridge-authz-usage in the `lightbridge-authz` repo) has been receiving zero access-log telemetry since https://github.com/ADORSYS-GIS/ai-helm/commit/948814de3157e1763e42fcfa4878df4203296810 (2026-05-31) repointed the Envoy access-log sink straight at Alloy and dropped the `-usage` collector that used to feed it. Its `usage_events` table has had no `account_id`-tagged rows landing from the gateway since. Source of truth for the restore requirement + the ingest routes this PR targets: https://github.com/ADORSYS-GIS/lightbridge-authz/blob/main/crates/lightbridge-authz-usage/src/routers/mod.rs and https://github.com/ADORSYS-GIS/lightbridge-authz/blob/main/AGENTS.md (usage-service section).

---

## 2. Intent

Restore the access-log feed to `lightbridge-authz-usage` **without disturbing the existing Alloy/Loki path** — the per-user Loki observability (ADR-0005) that `948814de` correctly preserved when it simplified the direct-to-Alloy sink must keep working exactly as today.

Two options were evaluated:

**(a) Add a second entry to `EnvoyProxy`'s own `accessLog.settings[].sinks[]`, pointing straight at the usage service.** Ruled out — not a design preference, a hard capability gap:
- Verified against `envoyproxy/gateway` `release/v1.8`'s `OpenTelemetryEnvoyProxyAccessLog` Go type (`api/v1alpha1/envoyproxy_accesslogging_types.go`): it only speaks **OTLP/gRPC** — `host`/`port` (default 4317), gRPC initial-metadata headers, `BackendCluster`-based TLS. There is **no URL-path field**, so it cannot target an OTLP/**HTTP** endpoint like `/v1/otel/logs`.
- `lightbridge-authz-usage`'s ingest is exactly the opposite: axum routes at `/v1/otel/{traces,metrics,logs}` (`crates/lightbridge-authz-usage/src/routers/mod.rs`), with **no gRPC receiver at all**.
- The two sides genuinely cannot speak to each other directly — this is the deciding factor, not a style choice.

**(b) Restore a dedicated `OpenTelemetryCollector`.** Chosen. The commit that removed the `-usage` collector also removed the `opentelemetry-operator` chart as "installed externally" — but this repo's own `CLAUDE.md` (and `charts/apps/values.yaml`'s comment) still document the OTel Operator as live in-cluster (`opentelemetry-system`, all `opentelemetry.io` CRDs present), reconciling the surviving `-traces` collector today. No operator reinstatement needed; this is a same-shaped resource next to one that's already running.

---

### In Scope

- Restoring the second (usage/billing) leg of the access-log fan-out via a collector.
- A same-namespace CA-trust `Certificate` for the collector, since the old `secretName: authz-tls` reference (pre-`948814de`) pointed at a Secret that lives in a **different** namespace (`converse`, owned by `lightbridge-authz-stack`) than this collector (`converse-gateway`) — and no cross-namespace secret replication (reflector, etc.) exists anywhere in `ai-helm` or `ai-helm-values`. Issuing a second cert off the same CA root and trusting only its `ca.crt` is the same pattern `ai-helm-values/environments/prod/deps/security-policies` already uses for Authorino trusting `lightbridge-opa`'s cert.

### Out of Scope

- The `ai-helm-values` `usage: enabled` flip (already merged on `main` as `ac34fa9`, by a parallel change) and anything else in that values repo.
- Any change to the Alloy/Loki/Tempo path, the `-traces` collector, or ADR-0005 per-user observability behavior.
- Re-adding the OpenTelemetry Operator chart (still external/already installed).

---

## 4. Verification

I verified this change by:

- [x] Running automated tests
- [ ] Running manual tests
- [ ] Checking logs
- [ ] Checking metrics
- [ ] Testing error cases
- [ ] Testing permissions/security behavior
- [ ] Testing rollback or failure behavior, if relevant

No live cluster access from this session — verification is `helm lint`/`helm template` only. Manual post-deploy verification is described below for whoever applies this.

Commands run:

```bash
cd charts/core-gateway
helm dependency update .
helm lint . --strict
helm template core-gateway . --namespace converse-gateway
helm template lightbridge-gateway . --namespace converse-gateway \
  -f <ai-helm-values>/environments/prod/values/core-gateway.yaml
```

Results:

```text
==> Linting .
[INFO] Chart.yaml: icon is recommended
1 chart(s) linted, 0 chart(s) failed
```

Rendered `EnvoyProxy.spec.telemetry.accessLog` sink (prod values):

```yaml
accessLog:
  settings:
    - sinks:
        - type: OpenTelemetry
          openTelemetry:
            host: lightbridge-gateway-core-gateway-usage.converse-gateway.svc.cluster.local
            port: 4317
            resources:
              k8s.cluster.name: "converse-llm-1"
              service.name: "envoy-ai-gateway"
      matches:
        - "request.headers['x-ai-eg-model'] != ''"
      # ...unchanged format.json block (all existing fields, byte-identical)
```

Rendered `-usage` `OpenTelemetryCollector` (prod values):

```yaml
apiVersion: opentelemetry.io/v1beta1
kind: OpenTelemetryCollector
metadata:
  name: lightbridge-gateway-core-gateway-usage
  namespace: converse-gateway
spec:
  mode: deployment
  volumes:
    - name: usage-ca
      secret:
        secretName: lightbridge-gateway-core-gateway-usage-ca
        items:
          - key: ca.crt
            path: ca.crt
  volumeMounts:
    - name: usage-ca
      mountPath: /etc/otel/tls
      readOnly: true
  config:
    receivers:
      otlp:
        protocols:
          grpc: { endpoint: 0.0.0.0:4317 }
          http: { endpoint: 0.0.0.0:4318 }
    processors:
      memory_limiter: { check_interval: 1s, limit_mib: 400, spike_limit_mib: 100 }
      batch: { send_batch_size: 512, timeout: 5s }
    exporters:
      otlphttp/lightbridge_usage:
        logs_endpoint: "https://lightbridge-usage.converse.svc.cluster.local:3000/v1/otel/logs"
        tls:
          ca_file: /etc/otel/tls/ca.crt
      otlp/alloy:
        endpoint: "alloy.observability.svc.cluster.local:4317"
        tls:
          insecure: true
    service:
      pipelines:
        logs:
          receivers: [otlp]
          processors: [memory_limiter, batch]
          exporters: [otlphttp/lightbridge_usage, otlp/alloy]
```

Plus the new `certificate-usage-ca.yaml` Certificate (`lightbridge-gateway-core-gateway-usage-ca`, `issuerRef: {name: self-signed-ca, kind: ClusterIssuer}`) and the untouched `-traces` collector, both rendered cleanly alongside it.

The **usage endpoint**: `https://lightbridge-usage.converse.svc.cluster.local:3000/v1/otel/logs` — service name/port/namespace confirmed by `helm template`-ing `lightbridge-authz`'s own `charts/lightbridge-authz-stack` against `ai-helm-values/environments/prod/values/lightbridge-app.yaml` (which enables `usage.enabled: true` as of `ac34fa9` on `main`), not assumed. **TLS**: real CA verification via the new `<fullname>-usage-ca` Certificate's `ca.crt` (same CA root as `lightbridge-authz-usage`'s own cert) — no `insecure`/skip-verify on that leg. **Alloy path**: unchanged endpoint (`alloy.observability.svc.cluster.local:4317`), unchanged `insecure: true` (that leg was never CA-verified, and this PR doesn't change that), unchanged JSON access-log format — only the hop changed (via the collector instead of direct), not the payload or destination.

---

## 5. Screenshots / Evidence

No screenshots (Helm chart change, not UI). Evidence is the rendered manifests in Section 4, and:

* Rendered EnvoyProxy diff, `-usage` collector, and `certificate-usage-ca.yaml`: this PR's diff.
* Prior state (what was removed): commit `948814de` ("chore(observability): remove usage-collector + otel-operator (external)").
* Endpoint confirmation: `lightbridge-authz` repo, `crates/lightbridge-authz-usage/src/routers/mod.rs` (`/v1/otel/logs` route) + `charts/lightbridge-authz-stack` rendered against `ai-helm-values/environments/prod/values/lightbridge-app.yaml` (service `lightbridge-usage`, namespace `converse`, port `3000`, HTTPS).

---

## 6. Risk Assessment

Risk level:

* [x] Low
* [ ] Medium
* [ ] High

Potential risks:

* The new `otlphttp/lightbridge_usage` exporter could back-pressure or error if `lightbridge-authz-usage` is slow/unreachable. Mitigated by `memory_limiter` (sheds load before OOM) and the fact that OTel Collector exporters fail independently per pipeline export — a failing `otlphttp` export doesn't block the `otlp/alloy` export in the same pipeline run (they're two exporters on one pipeline, not chained).
* If the CA cert doesn't propagate before the collector pod starts, `otlphttp/lightbridge_usage` exports would fail-closed with TLS errors (visible in collector logs) rather than silently accepting an unverified cert — since `insecure`/skip-verify was deliberately not used.
* No change to any resource the Alloy/Loki path depends on (same endpoint string, same `insecure: true`, same JSON format) — the only behavior change on that leg is an added collector hop.

Mitigation:

* Post-deploy: watch the `-usage` collector pod logs for exporter errors on both legs.
* Post-deploy: confirm Loki still receives per-user labels (query Grafana/Loki for `user_id`/`azp` as before) to prove the Alloy leg is unaffected.
* Post-deploy: query `usage_events` for rows with non-null `account_id` landing after the AI Gateway routes traffic, to prove the new leg is working:

  ```sql
  select count(*) from usage_events
  where account_id is not null
    and time > now() - interval '15 minutes';
  ```

---

## 7. AI Usage Declaration

AI was used for:

* [x] Understanding existing code
* [x] Generating code
* [ ] Refactoring
* [ ] Generating tests
* [x] Drafting documentation
* [x] Reviewing the diff
* [ ] Not used

Human verification:

* [ ] I understand every meaningful change in this PR
* [ ] I checked generated code manually
* [ ] I checked generated tests manually
* [ ] I removed unsupported AI assumptions
* [ ] I accept responsibility for this PR

This PR was authored by an AI agent (Claude) in a non-interactive session. The boxes above are left unchecked because they are human attestations — a human reviewer should verify this PR before merge, particularly the CA/TLS reasoning and the option (a)/(b) capability claim about `EnvoyProxy`'s access-log sink type.

---

## 8. Reviewer Focus

Please focus your review on:

* [x] Correctness
* [x] Architecture
* [x] Security
* [ ] Performance
* [ ] Tests
* [ ] Maintainability
* [x] Product intent
* [ ] Edge cases

Specifically: (1) is the `OpenTelemetryEnvoyProxyAccessLog` gRPC-only claim (Section 2) actually correct for the Envoy Gateway version this cluster runs; (2) is issuing a second `Certificate` off `self-signed-ca` the right way to get the collector a trusted CA, versus some cross-namespace secret mechanism this PR's author may have missed; (3) whether the `-usage` collector's resource sizing (same `memory_limiter`/`batch` tuning as `-traces`) is appropriate for access-log volume, which is likely much higher than trace volume.

### CODE REVIEW

File: charts/core-gateway/templates/otel.yaml [hunk 1]
```diff
@@ -49,8 +49,99 @@ spec:
...
+
+    exporters:
+      # lightbridge-authz-usage's OTLP/HTTP ingest only exposes a logs path
+      # (/v1/otel/logs — see crates/lightbridge-authz-usage/src/routers/mod.rs
+      # in the lightbridge-authz repo); there is no gRPC receiver on that
+      # service, which is exactly why this has to be `otlphttp` (path-aware)
+      # rather than a plain `otlp` gRPC exporter.
+      otlphttp/lightbridge_usage:
+        logs_endpoint: "https://lightbridge-usage.converse.svc.cluster.local:3000/v1/otel/logs"
+        tls:
```
[the remark this answers is not in the document]
Comment (stephane-segning) [reply charts/core-gateway/templates/otel.yaml:125]: Verified — this doesn't apply here, and I don't think a code change is needed on this PR.

**The mechanism you cite is correct** (`ca_file` without `insecure_skip_verify` does enforce hostname/SAN matching, not just chain trust), but the premise doesn't hold for the actual `authz-tls` cert:

`ai-helm-values/environments/prod/values/lightbridge-app.yaml:903-922` defines the real prod `authz-tls` Certificate (the one `lightbridge-authz-usage`'s TLS listener actually presents — this repo's own `global.tls.job` alternative is disabled in prod):

```yaml
commonName: lightbridge.{{ $.Release.Namespace }}.svc.cluster.local
dnsNames:
  - "*.{{ $.Release.Namespace }}.svc.cluster.local"
  - "*.{{ $.Release.Namespace }}.svc.cluster"
  - "*.{{ $.Release.Namespace }}.svc"
  - "*.{{ $.Release.Namespace }}"
```

With `$.Release.Namespace` = `converse` in prod, that's `*.converse.svc.cluster.local` — a **single-leftmost-label** wildcard. The `logs_endpoint` host here, `lightbridge-usage.converse.svc.cluster.local`, is exactly one label (`lightbridge-usage`) ahead of `converse.svc.cluster.local`, so it matches under standard wildcard-SAN matching (RFC 6125 leftmost-label-only; concretely, Go's `crypto/x509.Certificate.VerifyHostname` → `matchHostnames`, which the OTel Collector's `otlphttp` exporter uses, splits both pattern and host on `.` and only allows `*` to substitute the first label — that's satisfied here).

Also confirmed the Service name itself: `helm template` against `charts/lightbridge-authz-stack` (lightbridge-authz repo) with the `usage` alias renders a Service literally named `lightbridge-usage`, in namespace `converse` per the umbrella app's destination — matching the `logs_endpoint` host exactly, no typo.

So: no `server_name_override` needed, and no SAN gap to close. This PR's own files (`otel.yaml`, `certificate-usage-ca.yaml`) don't need changes — the risk would only be real if `authz-tls`'s dnsNames were ever narrowed to explicit hostnames instead of the namespace wildcard, which is a `ai-helm-values` concern, not this PR's. Leaving this thread as verified-safe rather than pushing a change.

(Meta note for the bot: good instinct to flag SAN vs CA-only verification as a real distinct failure mode — that's a legitimate class of bug. The gap is that the finding couldn't see the actual `authz-tls` Certificate resource, which lives in a different repo (`ai-helm-values`) from this diff, so it landed as "worth confirming" rather than being able to resolve itself. Correctly hedged as P2/"worth confirming" rather than a confident P0 assertion of breakage, which is the right call given the uncertainty — but the finding would be more actionable if it explicitly named the cross-repo location to check, i.e. "check `authz-tls`'s Certificate in ai-helm-values" rather than leaving verification entirely to the reviewer.)
